tools/lint_diff.ini

```diff
@@ -1,5 +1,5 @@
 [pycodestyle]
-max_line_length = 79
+max_line_length = 88
 statistics = True
-ignore = E121,E122,E123,E125,E126,E127,E128,E226,E241,E251,E265,E266,E302,E402,E704,E712,E721,E731,E741,W291,W293,W391,W503,W504
+ignore = E121,E122,E123,E125,E126,E127,E128,E226,E241,E251,E265,E266,E302,E402,E704,E712,E721,E731,E741,W291,W293,W391,W503,W504,E203,E501
```

Comment:
If [E501](https://www.flake8rules.com/rules/E501.html) is ignored, then `max_line_length` won't be enforced. Is this intentional?

Reply:
Good catch - this one is tricky. It's recommended that B950 is used instead of E501 with black, which I briefly mentioned 
[here](https://github.com/numpy/numpy/pull/21449/files/794131fbbef4d3bf464a70ec400da985b9684dfd#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449R64-R69) (but did not explain), and their team [talks about here](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length)).

So, the ideal case is to have B950 check everything since it's possible (but rare) that black will produce output that violates E501. But, B950 can't be added to the workflow that uses pycodestyle only, since it's in bugbear. This hook checks B950 for diffed lines but of course this hook isn't enforced on everything. Maybe a good option would be to set E501 to use line length 96, which would be B950's absolute max.

Reply:
Thanks for clarifying, was not aware of B950 being enforced as I was thinking from pycodestyle perspective. As it seems that `tox.ini` and `lint_diff.ini` has already diverged a bit, why not leave E501 here and as you mentioned in the other comment, when we move to pyflakes + bugbea, we can purge `lint_diff.ini`. What do you think?